lte/gateway/deploy/README.md

```diff
@@ -0,0 +1,237 @@
+### f. Update `/var/opt/magma/configs/control_proxy.yml` with your orc8r/controller details
+```bash
+sudo mkdir -p /var/opt/magma/configs/
+sudo vim /var/opt/magma/configs/control_proxy.yml
+
+cloud_address: controller.tchitchir.com
+cloud_port: 443
+bootstrap_address: bootstrapper-controller.tchitchir.com
+bootstrap_port: 443
+fluentd_address: fluentd.tchitchir.com
+fluentd_port: 443
+rootca_cert: /var/opt/magma/certs/rootCA.pem
+
+```
+### g. Update .env file
+```bash
+vim $MAGMA_ROOT/lte/gateway/docker/.env
+
+COMPOSE_PROJECT_NAME=agw
+DOCKER_USERNAME=
+DOCKER_PASSWORD=
+DOCKER_REGISTRY=
+IMAGE_VERSION=latest
+BUILD_CONTEXT=../../..
+ROOTCA_PATH=/var/opt/magma/certs/rootCA.pem
+CONTROL_PROXY_PATH=/var/opt/magma/configs/control_proxy.yml 
+SNOWFLAKE_PATH=/etc/snowflake
+CERTS_VOLUME=/var/opt/magma/certs
+CONFIGS_OVERRIDE_VOLUME=/var/opt/magma/configs
+CONFIGS_OVERRIDE_TMP_VOLUME=/var/opt/magma/tmp
+CONFIGS_DEFAULT_VOLUME=../configs
+CONFIGS_TEMPLATES_PATH=/etc/magma/templates
+LOG_DRIVER=journald
+```
+### h. Build AGW docker images 
+#### h.1 For ARM architecture
+```bash
+cd ${MAGMA_ROOT}/lte/gateway/docker/
+sudo docker-compose build --build-arg CPU_ARCH=aarch64 --build-arg DEB_PORT=arm64 --parallel
+```
+#### h.2 For X86 archtecture
+```bash
+cd ${MAGMA_ROOT}/lte/gateway/docker/
+sudo docker-compose build --parallel
+```
+
+### i. Start the services
+```bash
+cd ${MAGMA_ROOT}/lte/gateway/docker/
+docker-compose --env-file .env -f docker-compose.yaml -f docker-compose.dev.yaml up -d --force-recreate
+```
+
+**Note**: In case you want to use the docker images from the artifactory:
+You have to change directory to `/var/opt/magma/docker`
+
+```bash
+cd /var/opt/magma/docker
+vim .env
+
+COMPOSE_PROJECT_NAME=agw
+DOCKER_REGISTRY=docker.artifactory.magmacore.org/
+DOCKER_USERNAME=
+DOCKER_PASSWORD=
+IMAGE_VERSION=latest
+OPTIONAL_ARCH_POSTFIX=_arm
+ROOTCA_PATH=/var/opt/magma/certs/rootCA.pem
+CONTROL_PROXY_PATH=/var/opt/magma/configs/control_proxy.yml
+CONFIGS_TEMPLATES_PATH=/etc/magma/templates
+CERTS_VOLUME=/var/opt/magma/certs
+CONFIGS_OVERRIDE_VOLUME=/var/opt/magma/configs
+CONFIGS_OVERRIDE_TMP_VOLUME=/var/opt/magma/tmp
+CONFIGS_DEFAULT_VOLUME=../configs
+SECRETS_VOLUME=/var/opt/magma/secrets
+HOST_DOCKER_INTERNAL=192.168.129.1
+LOG_DRIVER=journald
+```
+
+After that you run the script `agw_upgrade.sh` or the compose file:
+```bash
+./agw_upgrade.sh
+```
+
+(or)
+
+```bash
+docker-compose down
+docker-compose --env-file .env -f docker-compose.yaml -f docker-compose.dev.yaml up -d --force-recreate
+```
+
+# II. Install S1apTester
+## 1. Pre-requisites
+### a. Clone the magma project
+```bash
+ubuntu@magma-trfserver:~$ git clone `https://github.com/magma/magma.git`
+```
+
+## 2. Run the Ansible Playbook
+To install the S1apTester, we need to run the Ansible Playbook `s1aptester.yml` with the inventory `hosts`
+```bash
+cd $MAGMA_ROOT/lte/gateway/deploy
+ansible-playbook -i hosts s1aptester.yml -K --key-file "~/.ssh/arm-key-pair-s1ap.pem"
```

Comment:
My virtual machine don't have the file arm-key-pair-s1ap.pem.
How can I find it or touch it?

Reply:
arm-key-pair-s1ap.pem is the key pair that you create on AWS for your instance(s), and you download to your environment, in my case I copy it to ~/.ssh